PWGHF/HFC/Tasks/taskCharmHadronsFemtoDream.cxx

```diff
@@ -303,14 +307,34 @@ struct HfTaskCharmHadronsFemtoDream {
   template <bool isMc, typename CollisionType, typename PartType, typename PartitionType1, typename PartitionType2, typename BinningType>
   void doMixedEvent(CollisionType const& cols, PartType const& parts, PartitionType1& part1, PartitionType2& part2, BinningType policy)
   {
-    processType = 1 << 1; // for mixed event
 
-    for (auto const& [collision1, collision2] : soa::selfCombinations(policy, mixingDepth.value, -1, cols, cols)) {
+    // Mixed events that contain at least one particle of interest
+
+    Partition<CollisionType> PartitionMaskedCol1 = (aod::femtodreamcollision::bitmaskTrackOne & BitMask) == BitMask && aod::femtodreamcollision::downsample == true;
+    PartitionMaskedCol1.bindTable(cols);
+    Partition<CollisionType> PartitionMaskedCol2 = (aod::femtodreamcollision::bitmaskTrackTwo & BitMask) == BitMask && aod::femtodreamcollision::downsample == true;
+    PartitionMaskedCol2.bindTable(cols);
+
+    processType = 1 << 1; // for mixed event
```

Comment:
More readable (or you can use `BIT(1)` if you prefer, but then I would use `BIT(0)` for the SE)
```suggestion
    processType = 2; // for mixed event
```